src/transformers/convert_slow_tokenizer.py

```diff
@@ -40,21 +39,24 @@ def __init__(self, model: str):
         self.sp = SentencePieceProcessor()
         self.sp.Load(model)
 
-    def extract(self) -> Tuple[Dict[str, int], List[Tuple]]:
+    def extract(self, vocab_scores=None) -> Tuple[Dict[str, int], List[Tuple]]:
         sp = self.sp
         vocab = {sp.id_to_piece(index): index for index in range(sp.GetPieceSize())}
+        if vocab_scores is not None:
+            vocab_scores, reverse = dict(vocab_scores), True
+        else:
+            vocab_scores, reverse = vocab, False
 
         # Merges
         merges = []
         for piece_l in vocab.keys():
             for piece_r in vocab.keys():
                 merge = f"{piece_l}{piece_r}"
-                piece_id = vocab.get(merge, None)
-                if piece_id:
-                    merges += [(piece_l, piece_r, piece_id)]
-        merges = sorted(merges, key=lambda val: val[2])
+                piece_score = vocab_scores.get(merge, None)
+                if piece_score:
+                    merges += [(piece_l, piece_r, piece_score)]
+        merges = sorted(merges, key=lambda val: val[2], reverse=reverse)
```

Comment:
This needs to be in its own PR with a flag for breaking change.

Reply:
It's not breaking anymore.

Reply:
It still has a very strong potential to be breaking as it touches code functionality, and it will be easier to isolate it in a git bisect if it goes in its own PR. So I insist.
You can just reopen the PR you closed and amend it with those changes.

Reply:
Fair enough.

Reply:
https://github.com/huggingface/transformers/pull/22582/files